### PR TITLE
[SD-2284][SD-2285] - Corrections for published stories

### PR DESCRIPTION
**Pull request by akintolga**

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/views/authoring-topbar.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <button class="btn btn-medium pull-left" ng-if="!sending && item._locked && can_unlock() && item.state !== 'published'" ng-click="unlock()" translate>Unlock</button>
         
         <button class="btn btn-medium pull-left" 
-            ng-show="!sending && !_editable && !item._locked && item.state !== 'published' && item.state !== 'scheduled' && publish_enabled" 
+            ng-show="!sending && !_editable && !item._locked && item.state !== 'published' && item.state !== 'corrected' && item.state !== 'killed' && item.state !== 'scheduled' && publish_enabled"
```
Comment (amagdas) [inline client/app/scripts/superdesk-authoring/views/authoring-topbar.html]: This is getting quite big, maybe we should create a property on the scope for it.

**Comment by amagdas** [comment]
:+1:

**Comment by marwoodandrew** [comment]
:+1:

**Comment by marwoodandrew** [comment]
Test might fail when validation PR is merged!

File: server/apps/publish/archive_publish.py [hunk 2]
```diff
@@ -65,6 +65,8 @@ class BasePublishService(BaseService):
     def on_update(self, updates, original):
         if not is_workflow_state_transition_valid(self.publish_type, original[app.config['CONTENT_STATE']]):
             raise InvalidStateTransitionError()
+        if original.get('item_id') and get_resource_service('published').is_published_before(original['item_id']):
+            raise PublishQueueError.post_publish_exists_error(Exception('Story with id:{}'.format(original['_id'])))
```
Comment (sivakuna-aap) [inline server/apps/publish/archive_publish.py]: Any specific reason for wrapping this error in `Exception`?

**Comment by akintolga** [comment]
Failing tests are fixed.

**Pull request merged**